go/vt/mysqlctl/builtinbackupengine.go

```diff
@@ -606,6 +606,17 @@ func (be *BuiltinBackupEngine) ExecuteRestore(ctx context.Context, params Restor
 // restoreFiles will copy all the files from the BackupStorage to the
 // right place.
 func (be *BuiltinBackupEngine) restoreFiles(ctx context.Context, params RestoreParams, bh backupstorage.BackupHandle, bm builtinBackupManifest) error {
+	// For optimization, we are replacing pargzip with pgzip, so newBuiltinDecompressor doesn't have to compare and print warning for every file
+	// since newBuiltinDecompressor is helper method and does not hold any state, it was hard to do it in that method itself.
+	if bm.CompressionEngine == PargzipCompressor {
+		originalEngineValue := bm.CompressionEngine
+		params.Logger.Warningf("engine \"pargzip\" doesn't support decompression, using \"pgzip\" instead")
+		bm.CompressionEngine = PgzipCompressor
+		defer func() {
+			bm.CompressionEngine = originalEngineValue
+		}()
```

Comment:
Do we even need the `originalEngineValue` as a variable at all? I mean we check that `bm.CompressionEngine` is equal to `PargzipCompressor`, so we could just use that value directly